bindings/node/lib/stateMachine.js

```diff
@@ -432,15 +432,16 @@ module.exports = function (modules) {
      */
     markCommand(client, ns, command, callback) {
       const bson = this.bson;
+      const options = { promoteLongs: false };
       const dbName = databaseNamespace(ns);
-      const rawCommand = bson.deserialize(command, { promoteLongs: false, promoteValues: false });
+      const rawCommand = bson.deserialize(command, options);
 
-      client.db(dbName).command(rawCommand, (err, response) => {
+      const commandOpts = rawCommand.encryptedFields ? options : {};
```

Comment:
Why is this conditional on encryptedFields usage?

Reply:
Because when I did it on all commands, our driver started having issues with values not being promoted where it expected it, or the types were getting promoted incorrectly. One example of this was cursor ids coming back as int32s instead of longs. This change is basically temporary until the server allows int/long to be interchangeable and then I'll revert this change.

Reply:
@durran Correct me if I'm wrong, but this flag controls only the decoding of  response for a request from libmongocrypt, back to libmongocrypt. Without this flag, we run the risk of changing the BSON types that libmongocrypt sees to be something other than what mongocryptd reported – isn't that a bug? and something that would mismatch the shared library behavior as well (where bson is passed through verbatim)?

If the time allows for it, I'd like to dig a bit deeper here.

Reply:
You are correct and I just brought it back for that case in the deserialization of the command and just use `{ promoteLongs: false }` on the subsequent command call.

Reply:
@durran What I’m having trouble with here is that there are commands that fail when you turn type promotion off for all command responses. That sounds very much like there’s a bug somewhere else that’s being masked by the promotion of values, and would prefer to dig into what that bug is. The expectation by libmongocrypt is generally that it will receive the *exact* response from mongocryptd. (I’m also happy to chat about this tomorrow in person if you prefer.)

> This change is basically temporary until the server allows int/long to be interchangeable and then I'll revert this change.

The point is that this shouldn’t happen; it’s a _good_ thing that int/long not being interchangeable exposes bugs here.

Reply:
@addaleax Yes let's chat tomorrow.